tests/integration/test_agent.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for jenkins-k8s-operator charm."""
+
+import logging
+from pathlib import Path
+
+import jenkinsapi.jenkins
+import requests
+import yaml
+from juju.application import Application
+from juju.model import Model
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text(encoding="utf-8"))
+
+
+async def test_jenkins_wizard_bypass(web_address: str):
+    """
+    arrange: given an active Jenkins charm's unit ip.
+    act: when web application is accessed
+    assert: wizard is bypassed and a login screen is shown.
+    """
+    response = requests.get(f"{web_address}/login", params={"from": "/"}, timeout=10)
+
+    assert "Unlock Jenkins" not in str(response.content)
+    assert "Welcome to Jenkins!" in str(response.content)
+
+
+async def test_jenkins_agent_relation(
+    model: Model,
+    application: Application,
+    jenkins_k8s_agent: Application,
+    jenkins_client: jenkinsapi.jenkins.Jenkins,
```

Comment:
I think we should have an integration test before that ensure the action is running properly to help in case of issues in the action that would fail this test
A simple call to jenkins api that would use this fixture before this test uses it

Reply:
The `jenkinsapi.jenkinis.Jenkins` on init, has a method that connects to the instance (hence the init would fail if there were any problems connecting to the instance). 
[Init](https://github.com/pycontribs/jenkinsapi/blob/ec2a23ff4d2ff539de34810da0a4189c6be1eb41/jenkinsapi/jenkins.py#L91) -> [poll] (https://github.com/pycontribs/jenkinsapi/blob/ec2a23ff4d2ff539de34810da0a4189c6be1eb41/jenkinsapi/jenkinsbase.py#LL40C18-L40C22) -> [python api request](https://github.com/pycontribs/jenkinsapi/blob/ec2a23ff4d2ff539de34810da0a4189c6be1eb41/jenkinsapi/jenkinsbase.py#L83).
Perhaps I'll document this behavior in the test so that it's less confusing! :D 
Thanks for the feedback!